synapse/rest/media/v1/config_resource.py

```diff
@@ -0,0 +1,37 @@
+# -*- coding: utf-8 -*-
+# Copyright 2018 Will Hunt <will@half-shot.uk>
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+#
+from twisted.web.server import NOT_DONE_YET
+from twisted.web.resource import Resource
+from synapse.http.server import respond_with_json, respond_with_json_bytes
+
+
+class MediaConfigResource(Resource):
+    isLeaf = True
+
+    def __init__(self, hs):
+        Resource.__init__(self)
+        config = hs.get_config()
+        self.limits_dict = {
+            "m.upload.size": config.max_upload_size,
+        }
+
+    def render_GET(self, request):
+        respond_with_json(request, 200, self.limits_dict, send_cors=True)
+        return NOT_DONE_YET
```

Comment:
Presumably we shouldn't be returning `NOT_DONE_YET` if we have actually finished the request.

@hawkowl how is this meant to work in twisted?

Reply:
I'm pretty sure this is correct. Your options for returning from a `render_*` method are:
- return a `bytes`, which twisted will then render as the body and call `request.finish()`
- return `NOT_DONE_YET`, which skips the above two.

the implication is that if you have rendered the body and called `finish()` yourself, you should call `NOT_DONE_YET`.

Note that this is exactly what happens with a whole bunch of our handlers that look like they are asynchronous, but actually complete synchronously.

HOWEVER. There *is* a problem here, in that `wrap_json_request_handler` does important things (1: adds exception handling, though that may be moot here; 2: adds logging). So, can haz `wrap_json_request_handler` please?